pkg/api/v1/firmware.go

```diff
@@ -0,0 +1,63 @@
+package serverservice
+
+import (
+	"github.com/google/uuid"
+	"github.com/volatiletech/null/v8"
+
+	"go.hollow.sh/serverservice/internal/models"
+)
+
+// ComponentFirmwareVersion represents a firmware file
+type ComponentFirmwareVersion struct {
+	UUID        uuid.UUID `json:"uuid"`
+	Vendor      string    `json:"vendor"`
+	Model       string    `json:"model"`
+	Filename    string    `json:"filename"`
+	Version     string    `json:"version"`
+	Component   string    `json:"component"`
+	Utility     string    `json:"utility"`
```

Comment:
Removed `Utility` on [ea58763](https://github.com/metal-toolbox/hollow-serverservice/pull/48/commits/ea58763668e66be101382ee6f20510bc9acb94cd). 

I think for `InstallMethod` runs into a similar issue as `Utility`. The firmware file can be installed with either, so maybe let's leave the decision to the installer component rather than adding it as part of the firmware file data.